command/server.go

```diff
@@ -1238,6 +1239,11 @@ CLUSTER_SYNTHESIS_COMPLETE:
 			if err := c.Reload(c.reloadFuncsLock, c.reloadFuncs, c.flagConfigs); err != nil {
 				c.UI.Error(fmt.Sprintf("Error(s) were encountered during reload: %s", err))
 			}
+
+		case <-c.SigUSR2Ch:
+			buf := make([]byte, 32*1024*1024)
```

Comment:
You could allocate this only the first time SigUSR2 is called and then just reslice on each call, but I don't think it's a super important optimization.